src/graph/GraphViewServer.ts

```diff
@@ -22,6 +22,10 @@ export class GraphViewServer extends EventEmitter {
   private _httpServer: http.Server;
   private _port: number | undefined;
   private _socket: SocketIO.Socket;
+  private _lastQuery: string | undefined;
```

Comment:
Can we turn these into a single object with members?  Something like
interface ClientViewState {
  query?: string;
  data?: any[];
  errorMessage?: string
}